nautobot/core/models/utils.py

```diff
@@ -170,3 +173,31 @@ def find_models_with_matching_fields(app_models, field_names, field_attributes=N
                 pass
     registry_items = {key: sorted(value) for key, value in registry_items.items()}
     return registry_items
+
+
+def construct_natural_key_slug(values):
+    """
+    Convert the given list of natural key values to a single URL-path-usable string.
+
+    - Non-URL-safe characters are percent-encoded.
+    - Null (`None`) values are percent-encoded as a literal null character `%00`.
+
+    Reversible by `deconstruct_natural_key_slug()`.
+    """
+    values = [str(value) if value is not None else "\0" for value in values]
+    values = NATURAL_KEY_SLUG_SEPARATOR.join(quote_plus(value, safe="/.:") for value in values)
```

Comment:
Consider **not** flagging `/` as a safe character (requiring it to be encoded/quoted instead) just to make it easier to use natural-key-slugs in URL patterns.

Reply:
We decided to quote/encode/escape the `/` character for now.